src/scalar/union.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+
+use crate::datatypes::DataType;
+
+use super::Scalar;
+
+/// A single entry of a [`crate::array::UnionArray`].
+#[derive(Debug, Clone, PartialEq)]
+pub struct UnionScalar {
+    value: Arc<dyn Scalar>,
+    type_: i8,
+    data_type: DataType,
+}
+
+impl UnionScalar {
+    /// Returns a new [`UnionScalar`]
+    #[inline]
+    pub fn new(data_type: DataType, type_: i8, value: Arc<dyn Scalar>) -> Self {
+        Self {
+            value,
+            type_,
+            data_type,
+        }
+    }
+
+    /// Returns the values irrespectively of the validity.
```

Comment:
```suggestion
    /// Returns the inner value
```

(there is no validity in the Union (and thus in the `UnionScalar`)

Reply:
Much clearer - committed it